### PR TITLE
Add backend code

### PR DESCRIPTION
**Pull request by n-wbrown**
This PR contains the backend Go code, alterations to the package specification to ensure that its loaded, some documentation, and CI scripts.

Happy to discuss it.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,10 @@
 {
+  "type": "datasource",
```
Comment (sasaki77) [inline package.json:2]: I'm not sure why `package.json` should be changed.
I think `src/plugin.json` contains these information already.

Comment (n-wbrown) [reply package.json]: @sasaki77, thank you for pointing this out. A correction has been pushed.

Comment (sasaki77) [reply package.json]: I'm sorry my ambiguous comment.
I think these lines are also unnecessary.

```
  "backend": true,
  "alerting": true,
  "executable": "gpx_archiver-datasource-backend",
```

What do you think?

Comment (n-wbrown) [reply package.json]: I agree

**Comment by klauer** [comment]
Very exciting to have this in. Great work, @n-wbrown! And thanks again for your openness to collaboration, @sasaki77. 

The front-end and back-end portions of this data source plugin will continue to be a vital tool for LCLS's operation monitoring with Grafana. 👏

**Pull request merged**